scipy/stats/_morestats.py

```diff
@@ -2901,15 +2904,20 @@ def bartlett(*samples, axis=0):
     if k < 2:
         raise ValueError("Must enter at least two input sample vectors.")
 
-    samples = _broadcast_arrays(samples, axis=axis, xp=xp)
-    samples = [xp.moveaxis(sample, axis, -1) for sample in samples]
+    if axis is None:
+        samples = [xp_ravel(sample) for sample in samples]
```

Comment:
Really, this is needed for any array-API backend. It is taken care of by the `_axis_nan_policy` decorator for NumPy, and eventually will probably be done for the other backends, too. But for now, it's simple to add it here.